pkg/command/root_initializers.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package command provides commands
+package command
+
+import (
+	"strings"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/centralconfig"
+	"github.com/vmware-tanzu/tanzu-cli/pkg/datastore"
+	"github.com/vmware-tanzu/tanzu-plugin-runtime/config"
+	configtypes "github.com/vmware-tanzu/tanzu-plugin-runtime/config/types"
+	"github.com/vmware-tanzu/tanzu-plugin-runtime/log"
+)
+
+const existingEndpointUpdateVersionKey = "existingEndpointUpdateVersion"
+
+// updateConfigWithTanzuPlatformEndpointChanges uses the central configuration to see if the endpoint
+// updates are needed for the CLI configuration or not. If it is needed start the endpoint updates
+// for all available tanzu contexts
+func updateConfigWithTanzuPlatformEndpointChanges() {
+	// Get the requested endpoint update version from the default central configuration
+	requestedEndpointUpdateVersion, err := centralconfig.DefaultCentralConfigReader.GetTanzuConfigEndpointUpdateVersion()
+	if err != nil || requestedEndpointUpdateVersion == "" {
+		return
+	}
+
+	// Get the existing endpoint update version from the datastore
+	existingEndpointUpdateVersion := ""
+	_ = datastore.GetDataStoreValue(existingEndpointUpdateVersionKey, &existingEndpointUpdateVersion)
+	if requestedEndpointUpdateVersion == existingEndpointUpdateVersion {
+		return
+	}
+
+	cfg, err := config.GetClientConfig()
+	if err != nil || cfg == nil || len(cfg.KnownContexts) == 0 {
+		return
+	}
+
+	// Get the endpoint update map from the default central configuration
+	endpointUpdateMap, err := centralconfig.DefaultCentralConfigReader.GetTanzuConfigEndpointUpdateMapping()
+	if err != nil {
+		return
+	}
+
+	updateSuccess := true
+	for idx := range cfg.KnownContexts {
+		ctx := cfg.KnownContexts[idx]
+		// If context is not eligible for endpoint update go to the next context
+		if !isValidContextForEndpointUpdates(ctx) {
+			continue
+		}
+
+		// Try to update endpoints in the tanzu context as per the endpointUpdateMap
+		updateEndpointsInTanzuContext(ctx, endpointUpdateMap)
+
+		// Update the context in the tanzu configuration file
+		if err := config.SetContext(ctx, false); err != nil {
+			updateSuccess = false
+		}
+	}
+
+	// if all the contexts are updated successfully, update the flag in the data store
+	if updateSuccess {
+		_ = datastore.SetDataStoreValue(existingEndpointUpdateVersionKey, &requestedEndpointUpdateVersion)
+		log.Info("The CLI contexts have been updated to use the new Tanzu Platform Endpoints.")
+	}
+}
+
+// isValidContextForEndpointUpdates validates if the specified context is valid tanzu context or not
+func isValidContextForEndpointUpdates(ctx *configtypes.Context) bool {
+	if ctx == nil || ctx.ContextType == configtypes.ContextTypeK8s {
+		return false
+	}
+	return true
```

Comment:
Yes. Actually, I wanted to support Tanzu and TMC context both. So, I have updated the method to return true if the contexttype is tanzu/tmc and return false otherwise.